### DISCUSSION TITLE
Docs FAQ: Why a monorepo?

### DISCUSSION THREAD

**Discussion by westhouseit** [directus/directus#3146] [Archive]
After a quick chat with Rijk on Discord I've had a few of my concerns over monorepo Directus cleared up. I thought it would be helpful for others to know the reasoning. Rijk suggested possibly an FAQ named as above and here's some of the best points from earlier:

**Comment by rijkvanzanten** [comment]
If you want to work on Directus without it being a mono repo, you have a way bigger problem.. 

All of the different parts of the project need to be cloned separately, and manually linked together with npm link. In case of a project as big as Directus that relies on multiple NPM packages, you can't really go without a monorepo..

Trying to do any dev work on v8 was pretty cumbersome, cause you had to clone 3 or 4 repos, run all things separately, and point them to each other in config etc. Right now, you can clone the repo once, run npm install and npm run dev in the root, and boom you're up and running!

As for "finer grained release tracking", this is still possible actually! In a monorepo, you can release the packages with individual versions. 
We actually made the conscious decision to release the sub-packages under the same version as the main project though. That way, you know for a fact what version of the SDK / gatsby plugin is supposed to work with what version of Directus.

If you have gatsby plugin v9.3.x you know for a fact it'll go with Directus v9.3.x. It gets confusing really fast if you have gatsby plugin v5.3 which only goes with 9.4+, but then isn't compatible with directus 9.5, cause that needs plugin v6 etc

I get your concerns though, I was wondering about the exact same things before we jumped ship
And then we did, and then after 2 weeks of fighting Lerna, and 2 months of use, it became obvious that this is the right decision for a project like Directus
